apps/desktop/src/lib/commits/utils.ts

```diff
@@ -0,0 +1,15 @@
+import type { DetailedCommit } from '$lib/vbranches/types';
+
+/**
+ * Find the last commit that diverged from the remote branch.
+ */
+export function findLastDivergentCommit(commits: DetailedCommit[]): DetailedCommit | undefined {
+	for (let i = commits.length - 1; i >= 0; i--) {
+		const commit = commits[i];
+		if (!commit) continue;
+		if (commit.id !== commit.remoteCommitId) {
+			return commit;
+		}
```

Comment:
I'm assuming the `if (!commit) ..` line makes eslint happy because of `noUncheckedIndexedAccess`, but imho we shouldn't need to do this. One solution could be using `const commit = commits[i] as DetailedCommit`.

Reply:
Yeah that's why haha. I don't mind the paranoid extra check, but I can remove it if you prefer 👍.

Also, a more concise alternative:
```ts
if (commit!.id !== commit!.remoteCommitId) {
    return commit;
}
```